vunit/ui/__init__.py

```diff
@@ -752,6 +762,47 @@ def _main(self, post_run):
         all_ok = self._main_run(post_run)
         return all_ok
 
+    def _update_test_filter(self, dependency_files):
+        """
+        Update test filter to match testbenches depending on user_files
+        """
+        # Find source file objects corresponding to the provided files.
+        # Non-existing files, directories and files not added to the
+        # project are ignored.
+        dependency_paths = []
+        for dependency_file in dependency_files:
+            if isinstance(dependency_file, str):
+                dependency_file = Path(dependency_file)
+            dependency_file = dependency_file.resolve()
+            if dependency_file.exists() and not dependency_file.is_dir():
```

Comment:
I think this is clearer, more efficient and more compact:

```suggestion
            if dependency_file.is_file():
```

Reply:
It includes the check that the path exists https://docs.python.org/3/library/pathlib.html#pathlib.Path.is_file